core/trino-main/src/main/java/io/trino/operator/join/SpilledLookupSourceHandle.java

```diff
@@ -70,7 +72,7 @@ public synchronized void setLookupSource(Supplier<LookupSource> lookupSource)
     {
         requireNonNull(lookupSource, "lookupSource is null");
 
-        if (state == State.DISPOSED) {
+        if (state == State.DISPOSE_REQUESTED || state == State.DISPOSE_COMPLETED) {
```

Comment:
Can it actually be in `DISPOSE_COMPLETED` state (since it's set by `HashBuilderOperator`)? IMO I don't think it can looking at `HashBuilderOperator#finish` loop (`setLookupSource` is called before `setDisposeCompleted`)

Reply:
nit: we still have `DISPOSE_COMPLETED` and `checkState` that state transitions are correct. But feel free to skip it